include/envoy/server/connection_handler.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+#include "envoy/network/connection.h"
+#include "envoy/network/filter.h"
+#include "envoy/network/listener.h"
+#include "envoy/network/listen_socket.h"
+#include "envoy/ssl/context.h"
+
+namespace Server {
```

Comment:
nit: Can we move this into network namespace. (Again not to cross streams since I don't think there is any need to). I would leave everything you did in the server implementation directory and just have it inherit from Network::ConnectionHandler